src/tests/integration/test_connection.py

```diff
@@ -199,3 +210,184 @@ def test_dbapi_connection_sets_right_dialect(dynamic_db, dialect, request):
 
     assert ConnectionManager.current.is_dbapi_connection
     assert ConnectionManager.current.dialect == dialect
+
+
+def test_duckdb_autocommit_on_with_manual_commit(tmp_empty, monkeypatch):
+    class Config:
+        autocommit = True
+
+    # monkeypatch.setattr(
+    #     connection, "set_sqlalchemy_isolation_level", Mock(return_value=False)
+    # )
+
+    engine = create_engine("duckdb:///my.db")
+
+    conn = SQLAlchemyConnection(engine=engine, config=Config)
+    conn_mock_commit = Mock(wraps=conn._connection.commit)
+    monkeypatch.setattr(conn._connection, "commit", conn_mock_commit)
+
+    conn.raw_execute(
+        """
+CREATE TABLE numbers (
+    x INTEGER
+);
+"""
+    )
+    conn.raw_execute(
+        """
+    INSERT INTO numbers VALUES (1), (2), (3);
+    """
+    )
+
+    # if commit is working, the table should be readable from another connection
+    another = SQLAlchemyConnection(
+        engine=create_engine("duckdb:///my.db"), config=Config
+    )
+    another_mock_commit = Mock(wraps=another._connection.commit)
+    monkeypatch.setattr(another._connection, "commit", another_mock_commit)
+
+    results = another.raw_execute("SELECT * FROM numbers")
+
+    assert list(results) == [(1,), (2,), (3,)]
```

Comment:
Do we have any function to easily iterate over these results returned by `raw_execute` (when multiple columns are returned)

Reply:
raw_execute returns an iterator, so you can do `[r for r in results]`. these objects are never used by end users since we create a ResultSet object, iterate there and return the ResultSet